src/ListView.vala

```diff
@@ -191,8 +191,49 @@ public class Tasks.ListView : Gtk.Grid {
         });
 
         editable_title.changed.connect (() => {
-            source.display_name = editable_title.text;
-            source.write.begin (null);
+            Tasks.Application.model.get_registry.begin ((obj, res) => {
+                try {
+                    var registry = Tasks.Application.model.get_registry.end (res);
+                    var collection_source = registry.find_extension (source, E.SOURCE_EXTENSION_COLLECTION);
+
+                    if (collection_source != null && source.has_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND)) {
+                        debug ("WebDAV Rename: %s", source.get_uid ());
+
+                        var source_webdav_extension = (E.SourceWebdav) source.get_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND);
+
+                        E.NamedParameters credentials;
+                        var credentials_provider = new E.SourceCredentialsProvider (registry);
+                        credentials_provider.lookup_sync (collection_source, null, out credentials);
+
+                        var collection_source_webdav_session = new E.WebDAVSession (collection_source);
+                        collection_source_webdav_session.credentials = credentials;
+
+                        var changes = new GLib.SList<E.WebDAVPropertyChange> ();
+                        changes.append (new E.WebDAVPropertyChange.set (
+                            E.WEBDAV_NS_DAV,
+                            "displayname",
+                            editable_title.text
+                        ));
+
+                        E.webdav_session_update_properties_sync (
+                            collection_source_webdav_session,
+                            source_webdav_extension.soup_uri.to_string (false),
+                            changes,
+                            null
+                        );
+
+                        registry.refresh_backend_sync (collection_source.uid, null);
+
+                    } else {
+                        debug ("Local Rename: %s", source.get_uid ());
+                        source.display_name = editable_title.text;
+                        registry.commit_source_sync (source, null);
+                    }
+
+                } catch (Error e) {
+                    critical (e.message);
```

Comment:
If we completely failed to rename the list, I think we should probably throw an error to the user